<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FittingDataDialog</class>
 <widget class="QDialog" name="FittingDataDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>760</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>加载观测数据配置</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QDialog, QWidget {
    background-color: #ffffff;
    color: #000000;
    font-family: &quot;Microsoft YaHei&quot;;
}
QGroupBox {
    border: 1px solid #cccccc;
    margin-top: 10px;
    padding-top: 10px;
    font-weight: bold;
}
QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top left;
    padding: 0 5px;
    color: #000000;
}
QLabel {
    color: #000000;
}
QLineEdit {
    border: 1px solid #999999;
    padding: 3px;
    background-color: #ffffff;
    color: #000000;
}
QPushButton {
    border: 1px solid #999999;
    background-color: #f0f0f0;
    padding: 5px 15px;
    border-radius: 3px;
    color: #000000;
}
QPushButton:hover {
    background-color: #e0e0e0;
}
QComboBox {
    border: 1px solid #999999;
    padding: 3px;
    background-color: #ffffff;
    color: #000000;
}
QComboBox::drop-down {
    border: 0px;
}
QTableWidget {
    border: 1px solid #cccccc;
    gridline-color: #eeeeee;
    color: #000000;
    background-color: #ffffff;
}
QHeaderView::section {
    background-color: #f0f0f0;
    color: #000000;
    border: 1px solid #cccccc;
    padding: 3px;
}
QRadioButton {
    color: #000000;
}
QCheckBox {
    color: #000000;
}
QSpinBox, QDoubleSpinBox {
    border: 1px solid #999999;
    background-color: #ffffff;
    color: #000000;
    padding: 2px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBoxSource">
     <property name="title">
      <string>1. 数据来源选择</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayoutSource">
        <item>
         <widget class="QRadioButton" name="radioProjectData">
          <property name="text">
           <string>从当前项目数据加载</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioExternalFile">
          <property name="text">
           <string>从外部文件导入 (*.csv, *.txt, *.xls)</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWidget" name="widgetFileSelect" native="true">
        <layout class="QHBoxLayout" name="horizontalLayoutFile">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="labelPath">
           <property name="text">
            <string>文件路径：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditFilePath">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnBrowse">
           <property name="text">
            <string>浏览...</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="labelPreview">
        <property name="text">
         <string>数据预览 (前 50 行)：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTableWidget" name="tablePreview">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxSettings">
     <property name="title">
      <string>2. 列映射与计算配置</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="horizontalSpacing">
       <number>15</number>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="labelTime">
        <property name="text">
         <string>时间列 (Time):</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboTime"/>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="labelPressure">
        <property name="text">
         <string>压力列 (Pressure):</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QComboBox" name="comboPressure"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelDerivative">
        <property name="text">
         <string>导数列 (Derivative):</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="comboDerivative"/>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="labelSkip">
        <property name="text">
         <string>跳过首行数:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSpinBox" name="spinSkipRows"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelTestType">
        <property name="text">
         <string>试井类型:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayoutTestType">
        <item>
         <widget class="QRadioButton" name="radioDrawdown">
          <property name="text">
           <string>压力降落 (Drawdown)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioBuildup">
          <property name="text">
           <string>压力恢复 (Buildup)</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelPi">
        <property name="text">
         <string>地层初始压力 (Pi):</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="spinPi">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="maximum">
         <double>9999.990000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="labelPiUnit">
        <property name="text">
         <string>MPa (仅降落试井需输入)</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: #666;</string>
        </property>
       </widget>
      </item>

      <item row="4" column="0">
       <widget class="QLabel" name="labelLSpacing">
        <property name="text">
         <string>导数计算参数 (L-Spacing):</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="spinLSpacing">
        <property name="minimum">
         <double>0.01</double>
        </property>
        <property name="maximum">
         <double>1.00</double>
        </property>
        <property name="singleStep">
         <double>0.01</double>
        </property>
        <property name="value">
         <double>0.10</double>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="labelLSpacingUnit">
        <property name="text">
         <string>(对数周期，默认 0.1)</string>
        </property>
        <property name="styleSheet">
         <string notr="true">color: #666;</string>
        </property>
       </widget>
      </item>

      <item row="5" column="0">
       <widget class="QLabel" name="labelSmooth">
        <property name="text">
         <string>导数平滑设置:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="5" column="1" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayoutSmooth">
        <item>
         <widget class="QCheckBox" name="checkSmoothing">
          <property name="text">
           <string>启用平滑 (Smoothing Span)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinSmoothSpan">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>99</number>
          </property>
          <property name="singleStep">
           <number>2</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>FittingDataDialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>FittingDataDialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>
